internal/status/status.go

```diff
@@ -31,10 +31,10 @@ import (
 	"errors"
 	"fmt"
 
-	"github.com/golang/protobuf/proto"
-	"github.com/golang/protobuf/ptypes"
 	spb "google.golang.org/genproto/googleapis/rpc/status"
 	"google.golang.org/grpc/codes"
+	"google.golang.org/protobuf/proto"
```

Comment:
From the previous PR: I'm still fairly sure this is a breaking API change.

The function `WithDetails` needs to now accept a `protoadapt.MessageV1` now instead of a `proto.Message`, because the new package's `proto.Message` is incompatible with the old one.

This would impact anyone using proto messages generated by some other tool, like (IIUC) `gogoproto`.  Or possibly(?) using a message generated by an old `protoc-gen-go`.

Reply:
@Clement-Jean -- This seems to be still unresolved?

Reply:
Weird.  It was fixed in this commit but now it's been reverted...

[`c784e0a` (#6919)](https://github.com/grpc/grpc-go/pull/6919/commits/c784e0a747bd0fba1a856f680da8b40f2654b78b)

Reply:
@Clement-Jean @dfawley Are we not resolving this issue? Things are breaking with this change.

Reply:
This PR is 6 months old now.  Please file a new issue with all the relevant details if you're seeing problems.

Reply:
@rajat-agarwal1 I'd be happy to help with the issue if you open a new issue and tag me